### PR TITLE
esp_tinyusb: cdc console feature is enabled only when VFS is enabled

### PR DESCRIPTION
**Pull request by esp-saurabhbansal**
cdc console feature is enabled only when VFS is enabled.
This is done by compiling the console feature related files only when VFS option is enabled.
If VFS is not enabled, compile error will be thrown. So, it will avoid runtime failure.

Closes https://github.com/espressif/idf-extra-components/issues/164

### CODE REVIEW

**Comment by esp-saurabhbansal** [comment]
@tore-espressif @roma-jam @Dazza0 
Please check this change.

**Comment by tore-espressif** [comment]
@esp-saurabhbansal the changes LGTM. 
Could you please change the commit message prefix from 'usbd' to 'esp_tinyusb' and add a second line to the commit message `Closes https://github.com/espressif/idf-extra-components/issues/164` ?

Could we add this commit to #178 and merge the changes together? This way we won't release 2 new versions (1.2.2 and 1.2.3) but only 1 new version (1.2.2)

File: usb/esp_tinyusb/CMakeLists.txt [hunk 1]
```diff
@@ -27,9 +27,13 @@ if(CONFIG_TINYUSB_CDC_ENABLED)
     list(APPEND srcs
         "cdc.c"
         "tusb_cdc_acm.c"
+        )
+if(CONFIG_VFS_SUPPORT_IO)
```
Comment (roma-jam) [inline usb/esp_tinyusb/CMakeLists.txt:31]: Nitpick: Probably, CONFIG_VFS_SUPPORT_IO_STREAMS would be little bit precisely. 
But it's up to you, indeed.

Comment (tore-espressif) [reply usb/esp_tinyusb/CMakeLists.txt:31]: This is coming from VFS component https://github.com/espressif/esp-idf/blob/master/components/vfs/Kconfig#L3

So we cannot change this :(

Comment (esp-saurabhbansal) [reply usb/esp_tinyusb/CMakeLists.txt:31]: CONFIG_VFS_SUPPORT_IO is a TOP level config option for VFS feature.

Comment (roma-jam) [reply usb/esp_tinyusb/CMakeLists.txt:31]: OK, I see.
I am not insisting. :slightly_smiling_face:

**Comment by esp-saurabhbansal** [comment]
> @esp-saurabhbansal the changes LGTM. Could you please change the commit message prefix from 'usbd' to 'esp_tinyusb' and add a second line to the commit message `Closes https://github.com/espressif/idf-extra-components/issues/164` ?
> 
> Could we add this commit to #178 and merge the changes together? This way we won't release 2 new versions (1.2.2 and 1.2.3) but only 1 new version (1.2.2)

This commit is added to PR #178

**Review by tore-espressif — APPROVED** [review]

**Pull request merged**